clicore/logger.go

```diff
@@ -0,0 +1,165 @@
+			// users can use the verbose flag to get full output to the console
+			fmt.Printf("Could not open log file %s for writing: %v\n", filepath.Join(pathElements...), err)
+		}
+		return nil
+	}
+
+	// if we decide we want to append logs, we can do it this way:
+	// (we would be required to first create the file and "rotate" it as it grows)
+	//file, err := os.OpenFile(path, os.O_WRONLY|os.O_APPEND, os.ModeAppend)
+
+	// for now, let's just create/overwrite the file each time
+	file, err := os.Create(path)
+	if err != nil {
+		if verboseMode {
+			// we don't want to return errors just because we cannot write logs to a file
+			// users can use the verbose flag to get full output to the console
+			fmt.Printf("Could not open log file %s for writing: %v\n", path, err)
+		}
+		return nil
+	}
+
+	// we want all messages to go to the file log
+	passAllMessages := zap.LevelEnablerFunc(func(lvl zapcore.Level) bool {
+		return true
+	})
+
+	// apply zap's lock and WriteSyncer helpers
+	fileDebug := zapcore.Lock(zapcore.AddSync(file))
+	if mockTargets != nil {
+		fileDebug = zapcore.Lock(mockTargets.FileLog)
+	}
+	fileLoggerEncoderConfig := soloPreferredEncoderConfig()
+	fileEncoder := zapcore.NewJSONEncoder(fileLoggerEncoderConfig)
+	fileCore := zapcore.NewCore(fileEncoder, fileDebug, passAllMessages)
+
+	return fileCore
+}
+
+func buildCliZapCoreConsoles(verboseMode bool, mockTargets *MockTargets) []zapcore.Core {
+	// force to verbose mode while running in a test environment
+	if mockTargets != nil {
+		verboseMode = true
+	}
+
+	// define error filter levels
+	errorMessages := zap.LevelEnablerFunc(func(lvl zapcore.Level) bool {
+		return lvl >= zapcore.ErrorLevel
+	})
+	stdOutMessages := zap.LevelEnablerFunc(func(lvl zapcore.Level) bool {
+		return lvl > zapcore.DebugLevel && lvl < zapcore.ErrorLevel
+	})
+	stdOutMessagesVerbose := zap.LevelEnablerFunc(func(lvl zapcore.Level) bool {
+		return lvl < zapcore.ErrorLevel
+	})
+
+	// add locks for safe concurrency
+	consoleInfo := zapcore.Lock(os.Stdout)
+	consoleErrors := zapcore.Lock(os.Stderr)
+	if mockTargets != nil {
+		consoleInfo = zapcore.Lock(mockTargets.Stdout)
+		consoleErrors = zapcore.Lock(mockTargets.Stderr)
+	}
+
+	consoleLoggerEncoderConfig := soloPreferredEncoderConfig()
+
+	// minimize the noise for non-verbose mode
+	if !verboseMode {
+		consoleLoggerEncoderConfig.EncodeTime = nil
+		consoleLoggerEncoderConfig.LevelKey = ""
+		consoleLoggerEncoderConfig.NameKey = ""
+	}
+	consoleEncoder := NewCliEncoder(CliLoggerKey)
+
+	consoleStdoutCore := zapcore.NewCore(consoleEncoder, consoleInfo, stdOutMessages)
+	if verboseMode {
+		consoleStdoutCore = zapcore.NewCore(consoleEncoder, consoleInfo, stdOutMessagesVerbose)
+	}
+	consoleErrCore := zapcore.NewCore(consoleEncoder, consoleErrors, errorMessages)
+	return []zapcore.Core{consoleStdoutCore, consoleErrCore}
+}
+
+func soloPreferredEncoderConfig() zapcore.EncoderConfig {
```

Comment:
nit: `defaultEncoderConfig()`

Current one seems overly verbose

Reply:
updated in https://github.com/solo-io/glooshot/pull/22